cmd/collector/main.go

```diff
@@ -2,21 +2,33 @@ package main
 
 import (
 	"context"
+	"flag"
 	"log"
 	"time"
 	"waffle/internal/packet"
 
 	"waffle/internal/worker"
 )
 
-const networkInterfaceDescription = "Intel(R) I211 Gigabit Network Connection"
+const (
+	defaultNetworkInterfaceDescription = "Intel(R) I211 Gigabit Network Connection"
+)
 
 func main() {
+	var (
+		networkInterface string
+	)
+	flag.StringVar(&networkInterface, "i", defaultNetworkInterfaceDescription, "Identification of the interface")
+
+	// question: Why do we need context here? It is not used in collector.Run, except of ctx.Done, but since it is not
+	// context.WithTimeout (as example) it can not be closed in any way.
+	// Same in c.serializer.SerializePackets(ctx, packetsChan), it can not be closed there as well.
+	// Why not just to remove it?
 	ctx := context.Background()
```

Comment:
Tbh I don't remember why I passed it there, you can remove it. Or, if you want to, you can implement some graceful shutdown on the application close for the sake of art 😄

Reply:
Yes, I will. Probably by capturing signals. I guess I will do another PR with graceful shutdown.

Reply:
It should be similar to this: 
```go
signal.Notify(quit, syscall.SIGINT, syscall.SIGTERM)

go func() {
        if err := srv.ListenAndServe(); err != nil && err != http.ErrServerClosed {
            log.Fatalf("Server failed: %s\n", err)
        }
    }()
log.Println("Server started on :8080")

<-quit
log.Println("Shutdown signal received, shutting down gracefully...")

ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
defer cancel()

if err := srv.Shutdown(ctx); err != nil {
        log.Fatalf("Server forced to shutdown: %s\n", err)
}
```